### PR TITLE
Add error listeners

### PR DESCRIPTION
**Pull request by peachbits**
The return values match SDK defaults since we haven't implemented any error handling

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208178889823029

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -31,23 +31,15 @@ export interface SpendFailure {
   errorCode?: string
 }
 
-export interface SynchronizerStatus {
+export interface StatusEvent {
```
Comment (samholmes) [inline src/types.ts:34]: Why is it called `StatusEvent` in this repo, but not in rn zcash?

File: src/types.ts [hunk 2]
```diff
@@ -31,23 +31,15 @@ export interface SpendFailure {
...
   alias: string
   name:
     | 'STOPPED' /** Indicates that [stop] has been called on this Synchronizer and it will no longer be used. */
     | 'DISCONNECTED' /** Indicates that this Synchronizer is disconnected from its lightwalletd server. When set, a UI element may want to turn red. */
-    | 'DOWNLOADING' /** Indicates that this Synchronizer is actively downloading new blocks from the server. */
-    | 'VALIDATING' /** Indicates that this Synchronizer is actively validating new blocks that were downloaded from the server. Blocks need to be verified before they are scanned. This confirms that each block is chain-sequential, thereby detecting missing blocks and reorgs. */
-    | 'SCANNING' /** Indicates that this Synchronizer is actively decrypting new blocks that were downloaded from the server. */
-    | 'ENHANCING' /** Indicates that this Synchronizer is actively enhancing newly scanned blocks with additional transaction details, fetched from the server. */
```
Comment (samholmes) [inline src/types.ts:42]: We don't want these event name types?

File: android/src/main/java/app/edge/rnpiratechain/RNPiratechainModule.kt [hunk 3]
```diff
@@ -86,12 +86,46 @@ class RNPiratechainModule(private val reactContext: ReactApplicationContext) :
...
+                true
+            }
+            wallet.onSetupErrorHandler = { error ->
+                handleError("error", error)
+                false
+            }
+            wallet.onSubmissionErrorHandler = { error ->
+                handleError("error", error)
+                false
+            }
```
Comment (samholmes) [inline android/src/main/java/app/edge/rnpiratechain/RNPiratechainModule.kt:123]: Is `error` always a string?

**Review by samholmes — CHANGES_REQUESTED** [review]
Just some questions (also maybe answer before merging the react-native-zcash PR just in case)

File: src/types.ts (hunk 2 above)
Comment (peachbits) [reply src/types.ts:42]: Older versions of the SDK used them but they don't exist anymore

File: android/src/main/java/app/edge/rnpiratechain/RNPiratechainModule.kt (hunk 3 above)
Comment (peachbits) [reply android/src/main/java/app/edge/rnpiratechain/RNPiratechainModule.kt:123]: `error` is a `Throwable?`

Comment (samholmes) [reply android/src/main/java/app/edge/rnpiratechain/RNPiratechainModule.kt:123]: Is a `Throwable?` serializable to a string? I'm wondering if there is a chance the kotlin code could slip some other non-string kind of data type to the JS environment thus making the `message: string` type a lie.

If you're confident that this couldn't happen, then approved.

**Review by samholmes — APPROVED** [review]

**Pull request merged**